SolutionsAndTests

```diff
@@ -0,0 +1,258 @@
+function problem1(numbers)
+{
+	// safety checks, test for param null or not an array
+	if(	numbers == null)
+	{
+		return 0;
+	}
+	if(!(Object.prototype.toString.call( numbers ) === '[object Array]' ))
+	{
+		return 0;
+	}
```

Comment:
I don't think these safety checks are required. Given the fact that the specifications of the problem say that you are given an array, I think you should go ahead and make the calculations. As a matter of fact, these kind of pre-emptive checks might hide errors in a bigger system where for example a component should pass an array to another component but it actually passes a null value. In these cases, I think, your code **should** crash or at least raise a warning to signal that something is wrong and not just accept the invalid value.